lib/LanguageServerCompletion/Protocol/InsertText.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerCompletion\Protocol;
+
+use LanguageServerProtocol\InsertTextFormat;
+
+final class InsertText
+{
+    /**
+     * @var string|null
+     */
+    private $value;
+
+    /**
+     * @var int|null
+     */
+    private $type;
+
+    public function __construct(?string $value, ?int $type = null)
```

Comment:
this should be private IMO if you're using named constructors (so no `null` values for the type)

Reply:
For me they are not exclusive, I also do `new InsertText(null)`.
I could add `InsertText::createEmpty()` or `InsertText::none()` but I'm not sure about the why ?

But I would love to know why so don't hesitate to expand on that ;)

Anyway I won't do it quit soon, after talking with @dantleech we decided to move a big part of this to the completion package and deal with it in `Suggestion` instead.
So I will rewrite this PR once this will be done.